Makefile

```diff
@@ -1,13 +1,11 @@
 PROJECT_ROOT=github.com/jaegertracing/jaeger
 # TOP_PKGS is used with 'go test'
-# TODO: try to do this without glide, since it may not be installed initially
-TOP_PKGS := $(shell glide novendor | \
-	sort | \
+TOP_PKGS := $(shell go list ./... | \
 	grep -v \
-		-e ./thrift-gen/... \
-		-e ./swagger-gen/... \
-		-e ./examples/... \
-		-e ./scripts/...\
+		-e examples \
+		-e scripts \
+		-e swagger-gen \
+		-e thrift-gen\
```

Comment:
this is not equivalent, glide only returns top-level pkgs
```
$ glide novendor
./swagger-gen/...
./crossdock/...
./cmd/...
./plugin/...
./storage/...
./thrift-gen/...
./examples/...
./model/...
./pkg/...
.
```

Reply:
Well, they are returning different result. But once they go to `go test $TOP_PKGS`, the test results are same.
There are two options I have in mind:
1. Keep using glide novendor
2. Rename `TOP_PKGS` to `TEST_PKGS`

Reply:
I certainly want to stop using glide for that, I'm just concerned that there are 8 occurrences of `TOP_PKGS` in the Makefile, so we need to ensure that changing what it refers to is still equivalent for all other built steps using it.

Reply:
Got it. I understand your concern.

`TOP_PKGS` is used by 4 Go tools: `go list`, `go test`, `go vet`, and `golint`. And 2 other tools: `gosec` and `gosimple`. All of Go tools have the same behavior on handling packages argument. I also read `gosec` and `gosimple` documentation and the way they handle packages argument is same as Go tools.

Reading documentation is not enough, so I created a simple script to compare them. Please check it here: https://gist.github.com/ariefrahmansyah/89cecbd7f933f80a9258c0028c92a2f3

I'm sure that migrating from `glide novendor` to `go list` is fine 🙂 

By the way, another thing to discuss, maybe we can merge `TOP_PKGS` to `ALL_PKGS` 🤔

Reply:
> By the way, another thing to discuss, maybe we can merge TOP_PKGS to ALL_PKGS

yes, I think it would make sense, I never liked the split, it was only there because some tools worked differently.